### PR TITLE
#41 More filters for finding participants

### LINKED ISSUE #41: More filters for finding participants
Currently, there are only five filters for finding participants. However, the GitHub API can access significantly more information than what is currently being used. Add as many filters as possible to more accurately reflect the information available.

### PR DESCRIPTION
**Pull request by oliver-allen**
This pull request closes #41 

There are 3 new properties to filter a participant by:
- Name
- Number of followers
- Number of following

The Participant class now has these properties as fields and can be created through the GitHub API and manually creating a participant.

The Participant page includes extra fields for filtering participants by these properties. They are also reflected in the table of participants
screen shot 2018-03-28 at 5 14 02 pm

The viewing and editing pages of Participants also reflect these new properties
screen shot 2018-03-28 at 5 20 08 pm screen shot 2018-03-28 at 5 20 24 pm

### Related Issues Complete
Limit quantitative participant filtering #117 limited the range for the number of contributions and number of repositories. The new followers and following filters are also limited to the range from 0 - max value

### Tests
The unit tests relating to participants have been updated to test the new Participant

### CODE REVIEW

**Comment by oliver-allen** [comment]
This will effects some of the open issues:

#34 Link filtering participants to studies
#136 Adding unit tests for GitHubResource REST controller and refactoring for Testability
#149 Bug: Only allow positive numeric values of a participant

**Review by xavier630 — APPROVED** [review]
Nice one! We did the review together at your computer.

**Review by JoeHorvath — CHANGES_REQUESTED** [review]
In Participant.json it looks like you put 'followers' in twice. Was one of them meant to be 'following'?

File: .jhipster/Participant.json [hunk 1]
```diff
@@ -36,11 +40,19 @@
         {
             "fieldName": "numberOfRepositories",
             "fieldType": "Integer"
+        },
+        {
+            "fieldName": "followers",
```
Comment (JoeHorvath) [inline .jhipster/Participant.json]: Should be "following"

**Comment by oliver-allen** [comment]
Good splotting. It seems to work even with that error. Fixed and tested.

**Review by JoeHorvath — APPROVED** [review]
Nice work. Remember to squash your commits!

**Comment by oliver-allen** [comment]
@softeng-701 Ready to merge. It has been reviewed and squashed as much as I can

**Pull request closed without merging**